decision-records/2025-04-16--redaction-processor-for-otelcol.md

```diff
@@ -0,0 +1,160 @@
+**Date:** 2025-04-16
+**Author:** Jose Massón (@Abuelodelanda)
+
+
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [Context and Problem Statement](#context-and-problem-statement)
+- [Decision: Redaction Processor](#decision-redaction-processor)
+  - [Config examples](#config-examples)
+    - [Log record in plain text format](#log-record-in-plain-text-format)
+    - [Log record in JSON format](#log-record-in-json-format)
+    - [Use these configs in a `pipeline`](#use-these-configs-in-a-pipeline)
+  - [How Loki stores the redacted logs:](#how-loki-stores-the-redacted-logs)
+    - [Log record in plain text format:](#log-record-in-plain-text-format)
+    - [Log record in JSON format:](#log-record-in-json-format)
+- [Interaction through `juju` command line interface](#interaction-through-juju-command-line-interface)
+
+<!-- markdown-toc end -->
+
+
+
+## Context and Problem Statement
+
+For applications that deal with sensitive, private data, the ability to mask certain fields or regular expressions in their logs is important, in some cases required. As OpenTelemetry Collector charm deployments will serve as the entry points for log data, they should support log masking before they send logs to Loki (or another destination).
+
+This log masking feature should allow for specifying which field values to mask in the case of structured logs and regular expressions to use for masking parts of unstructurted log lines. These should be configurable through the charm libraries, relations and / or through Juju config.
+
+OpenTelemetry Collector charms support masking logs before they are shipped to Loki (or another destination) with fields / regex specified through configuration.
+
+
+## Decision: Redaction Processor
+
+We will use the [redaction processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/redactionprocessor) which will be enabled in the otelcol binaries we ship since it is specifically designed to prevent sensitive fields from leaking into your telemetry data. It offers powerful tools to:
```

Comment:
This doesn't look like a decision but like a feature request or task description.
If there's no choice there's no decision, and I am missing something like "rejected alternatives" or a discussion of what the problematic aspect of the issue is, that required an ADR.